coalals/main.py

```diff
@@ -0,0 +1,56 @@
+import sys
+from socketserver import TCPServer
+from argparse import ArgumentParser
+
+from coala_utils.decorators import enforce_signature
+from .utils.wrappers import StreamHandlerWrapper
+from .langserver import LangServer
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+@enforce_signature
+def start_tcp_lang_server(handler_class: LangServer, bind_addr, port):
+    wrapper_class = type(
+        handler_class.__name__ + 'Handler',
+        (StreamHandlerWrapper,),
+        {'DELEGATE_CLASS': handler_class, },
+    )
+
+    try:
+        server = TCPServer((bind_addr, port), wrapper_class)
+    except Exception as e:
```

Comment:
Since you are going to run this from CLI I think you should catch a `KeyboardInterrupt` separately and just `exit('Command killed by keyboard interrupt')` or something like that.

Reply:
Yes, the TCP mode will mostly be used from CLI during debugging.